File: php-sdk/src/Api/BlacklistApi.php
<?php

declare(strict_types=1);

namespace Cleanster\Api;

use Cleanster\HttpClient;
use Cleanster\Models\ApiResponse;

/**
 * Manages the list of cleaners blocked from auto-assignment.
 */
final class BlacklistApi
{
    public function __construct(private readonly HttpClient $http) {}

    /** Return all cleaners currently on the blacklist. */
    public function listBlacklistedCleaners(): ApiResponse
    {
        $raw = $this->http->get('/v1/blacklist/cleaner');
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }

    /**
     * Prevent a cleaner from being auto-assigned to bookings.
     *
     * @param int         $cleanerId Cleaner to blacklist.
     * @param string|null $reason    Optional reason.
     */
    public function addToBlacklist(int $cleanerId, ?string $reason = null): ApiResponse
    {
        $body = ['cleanerId' => $cleanerId];
        if ($reason !== null && $reason !== '') {
            $body['reason'] = $reason;
        }
        $raw = $this->http->post('/v1/blacklist/cleaner', $body);
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }

    /** Re-enable a previously blacklisted cleaner for auto-assignment. */
    public function removeFromBlacklist(int $cleanerId): ApiResponse
    {
        $raw = $this->http->delete('/v1/blacklist/cleaner', ['cleanerId' => $cleanerId]);
        return new ApiResponse($raw['status'] ?? 200, $raw['message'] ?? 'OK', $raw['data'] ?? []);
    }
}

File: ruby-sdk/lib/cleanster/api/blacklist_api.rb
module Cleanster
  module Api
    # BlacklistApi — prevent specific cleaners from being assigned to your bookings.
    class BlacklistApi
      def initialize(http)
        @http = http
      end

      # Return all cleaners currently on the blacklist.
      # @return [Models::ApiResponse]
      def list_blacklisted_cleaners
        raw = @http.get("/v1/blacklist/cleaner")
        Models::ApiResponse.from_hash(raw)
      end

      # Add a cleaner to the blacklist.
      #
      # @param cleaner_id [Integer]
      # @param reason     [String, nil] Optional reason.
      # @return [Models::ApiResponse]
      def add_to_blacklist(cleaner_id:, reason: nil)
        body = { cleanerId: cleaner_id }
        body[:reason] = reason if reason
        raw = @http.post("/v1/blacklist/cleaner", body: body)
        Models::ApiResponse.from_hash(raw)
      end

      # Remove a cleaner from the blacklist.
      #
      # @param cleaner_id [Integer]
      # @return [Models::ApiResponse]
      def remove_from_blacklist(cleaner_id:)
        raw = @http.delete("/v1/blacklist/cleaner", body: { cleanerId: cleaner_id })
        Models::ApiResponse.from_hash(raw)
      end
    end
  end
end
